a1/src/main/java/ro/utcn/spet/a1/controller/ConsoleController.java

```diff
@@ -0,0 +1,142 @@
+package ro.utcn.spet.a1.controller;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.stereotype.Component;
+import ro.utcn.spet.a1.exception.QuestionNotFoundException;
+import ro.utcn.spet.a1.exception.TagNotFoundException;
+import ro.utcn.spet.a1.model.Question;
+import ro.utcn.spet.a1.model.Tag;
+import ro.utcn.spet.a1.model.User;
+import ro.utcn.spet.a1.service.QuestionService;
+import ro.utcn.spet.a1.service.TagService;
+import ro.utcn.spet.a1.service.UserService;
+
+import java.util.Scanner;
+
+
+@Component
+@RequiredArgsConstructor
+public class ConsoleController implements CommandLineRunner {
+    private final Scanner scanner = new Scanner(System.in);
+    private final UserService userService;
+    private final QuestionService questionService;
+    private final TagService tagService;
+    private User user;
+    @Override
+    public void run(String... args) throws Exception {
+        print("Welcome!");
+        boolean done = false;
+        while (!done) {
+            print("Enter command: ");
+            String command = scanner.next().trim();
+            try {
+                done = handleCommand(command);
+            } catch (QuestionNotFoundException questionNotFoundException) {
+                print("The student with the given ID was not found!");
+            }
+
+        }
+    }
+
+    private boolean handleCommand(String command) {
+        switch (command) {
+            case "login":
+                handleLogin();
+                return false;
+            case "list":
+                handleList();
+                return false;
+            case "add":
+                handleAdd();
+                return false;
+            case "remove":
+                handleRemove();
+                return false;
+            case "filtertext":
+                handleFilterText();
+                return false;
+            case "filtertag":
+                handleFilterTag();
+                return false;
+            case "exit":
+                return true;
+            default:
+                print("Unknown command. Try again.");
+                return false;
+        }
+    }
+
+    private void handleLogin() {
+        print("Username:");
+        String username = scanner.next().trim();
+        print("password:");
+        String password = scanner.next().trim();
+        this.user = userService.validateUser(username, password);
+        if(null != user) {
+            print("Welcome user: " + user.getUsername() + ".");
+        }else{
```

Comment:
Use your IDE to reformat your code, e.g. https://www.jetbrains.com/help/idea/using-code-editor.html#reformat_rearrange_code. This will add spaces and wrap the code appropriately.